File: 11.container-with-most-water/solution.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public int maxArea(int[] height) {
        Integer[] arr = new Integer[height.length];
		for (int i = 0; i < arr.length; i++)
			arr[i] = i;
		Arrays.sort(arr, new Comparator<Integer>(){
			public int compare(Integer o1, Integer o2)
			{
				int value = Integer.compare(height[o2], height[o1]);
				return value == 0 ? Integer.compare(o2, o1) : value;
			}
		});
		int left = Integer.MAX_VALUE;
		int right = Integer.MIN_VALUE;
		int ans = 0;
		for (int i : arr)
		{
			if (left == Integer.MAX_VALUE)
			{
				left = i;
				right = i;
			} else
			{
				ans = Math.max(ans, height[i] * Math.max(i - left, right - i));
				left = Math.min(left, i);
				right = Math.max(right, i);
			}
		}
		return ans;
    }
}

File: 11.container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i = 0
        j = len(height) - 1
        val = 0
        while i < j:
            val = max(val, (j - i) * min(height[i], height[j]))
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return val
